lib/galaxy/tools/data_fetch.py

```diff
@@ -254,14 +255,15 @@ def _resolve_item_with_primary(item):
             hash_value = item.get(hash_function)
             if hash_value:
                 hashes.append({"hash_function": hash_function, "hash_value": hash_value})
-        for hash_dict in hashes:
-            hash_function = hash_dict.get("hash_function")
-            hash_value = hash_dict.get("hash_value")
-            try:
-                _handle_hash_validation(upload_config, hash_function, hash_value, path)
-            except Exception as e:
-                error_message = str(e)
-                item["error_message"] = error_message
+        if path:
```

Comment:
This might be obvious.. but why do we validate hashes only when path is not none?

Reply:
`path` is None when the dataset is deferred (see lines 243-246), so the hashes can only be validated when the dataset is materialized.